darshan-util/pydarshan/darshan/experimental/plots/plot_opcounts.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+
+from darshan.report import *
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def plot_opcounts(report):
+    """
+    Generates a bar chart summary for operation counts.
+
+	Args:
+    	report (DarshanReport): darshan report object to plot
+    """
+
+    # defaults
+    labels = ['Read', 'Write', 'Open', 'Stat', 'Seek', 'Mmap', 'Fsync']
+    posix_vals = [0, 0, 0, 0, 0, 0, 0]
+    mpiind_vals = [0, 0, 0, 0, 0, 0, 0]
+    mpicol_vals = [0, 0, 0, 0, 0, 0, 0]
+    stdio_vals = [0, 0, 0, 0, 0, 0, 0]
+
+
+    # TODO: change to report.summary
+    if 'agg_ioops' in dir(report):
+        print("Summarizing... agg_ioops")
+        report.agg_ioops()
+    else:
+        print("Can not create summary, agg_ioops aggregator is not registered with the report clase.")
+
+
+    mods = report.summary['agg_ioops']
```

Comment:
I haven't looked at the `agg_ioops` stuff in detail, but if the team prefers to use what is there I don't really mind. I do note that at the moment, this errors out for me even with a simple file and I'm not entirely sure if `DXT` handling is available for the control flow in this function:

```python
import darshan
from darshan.experimental.plots.plot_opcounts import plot_opcounts

logfile = '/home/tyler/github_projects/darshan/darshan-util/pydarshan/tests/input/sample-dxt-simple.darshan'
report = darshan.DarshanReport(logfile, read_all=True)
ret = plot_opcounts(report=report)
```

```python
Traceback (most recent call last):
  File "test_jakob_opcounts.py", line 6, in <module>
    ret = plot_opcounts(report=report)
  File "/home/tyler/python_37_venv/lib/python3.7/site-packages/darshan-3.3.0.3-py3.7.egg/darshan/experimental/plots/plot_opcounts.py", line 34, in plot_opcounts
    mods = report.summary['agg_ioops']
KeyError: 'agg_ioops'
```

Regardless, we'll probably want a clear error message--at the moment something gets printed, but not sure what it means.

`Can not create summary, agg_ioops aggregator is not registered with the report clase.`

For example, if it is having a problem specifically with `DXT` it should say so. If it is something else that is wrong/missing, a suggestion might be helpful for the developer/end user.

Reply:
The report.summary namespace is something I am using in some of the interactive analysis tools to cache already performed aggregations. We could think about breaking these out to somehwere else, but as far this pull request goes, it just takes the status quo and splits it into multiple files. So I think this should be addressed seperately.